cms/forms.py

```diff
@@ -16,3 +21,85 @@ def __init__(self, data=None, files=None, parent_page=None, *args, **kwargs):
             self.fields["overrides"].block.child_blocks["course_run"].child_blocks[
                 "readable_id"
             ].parent_readable_id = parent_page.specific.course.readable_id
+
+
+class CoursePageForm(WagtailAdminPageForm):
+    """
+    Admin form for CoursePage and ExternalCoursePage.
+
+    This form introduces price and course_run fields to manage pricing in cms.
+    """
+
+    price = forms.DecimalField(required=False)
+    course_run = forms.ChoiceField(required=False)
+
+    def __init__(self, data=None, files=None, parent_page=None, *args, **kwargs):
+        """
+        Adds choices for course_run field.
+        """
+        super().__init__(data, files, parent_page, *args, **kwargs)
+
+        instance = kwargs.get("instance", None)
+        if instance and instance.id:
+            course_runs = instance.course.courseruns.all()
+            course_run_choices = [("", "")] + [(run.id, run) for run in course_runs]
+            self.fields["course_run"].choices = course_run_choices
+
+    def save(self, commit=True):
+        """
+        Handles pricing update and creates product(if required) and product version for a course run.
+        """
+        page = super().save(commit=False)
+
+        course_run_id = self.cleaned_data["course_run"]
+        price = self.cleaned_data["price"]
+        if course_run_id and price:
+            course_run = CourseRun.objects.get(id=course_run_id)
+            product, _ = Product.objects.get_or_create(
+                content_type=ContentType.objects.get_for_model(CourseRun),
+                object_id=course_run.id,
+            )
+            ProductVersion.objects.create(product=product, price=price)
+
+        if commit:
+            page.save()
+        return page
+
+
+class ProgramPageForm(WagtailAdminPageForm):
+    """
+    Admin form for ProgramPage and ExternalProgramPage.
+
+    This form introduces price field to manage pricing in cms.
+    """
+
+    price = forms.DecimalField(required=False)
+
+    def __init__(self, data=None, files=None, parent_page=None, *args, **kwargs):
+        """
+        Initialize the price field with the existing program price.
+        """
+        from cms.models import ProgramPage, ExternalProgramPage
+
+        super().__init__(data, files, parent_page, *args, **kwargs)
+        instance = kwargs.get("instance", None)
+        if instance and instance.id:
+            self.fields["price"].initial = instance.program.current_price
```

Comment:
I see an unexpected `0.0` when the price is auto set
<img width="992" alt="image" src="https://github.com/mitodl/mitxpro/assets/34372316/635f45ce-b7f6-4188-9375-c8ad8fc2dce8">

Reply:
It is the wagtail icon for the decimal field.